dash/backend/src/modules/shared/services/mine-logger.service.ts

```diff
@@ -21,42 +21,52 @@ export class MineLoggerService extends ConsoleLogger {
+      } else if (typeof trace === 'object' && trace instanceof Error) {
+        errorTrace = trace.stack;
+      }
 
-    super.error(message, errorTrace, context);
-    const messageLabel = typeof message === 'object' ? message?.label : message;
+      super.error(message, errorTrace, context);
+      const messageLabel = typeof message === 'object' ? message?.label : message;
 
-    if (
-      !(this.messagesToIgnore.includes(messageLabel))  // the message isn't in the array of messages to skip sending emails for
-      && this.configService.get('email.system.enableErrorReportEmail')  // and error report emails are enabled
-    ) {
-      const now = new Date().toUTCString();
-      this.mailerService.sendMail({
-        to: this.configService.get('email.system.errorReportReceiver'),
-        from: this.configService.get('email.default.sender'),
-        subject: `m9sweeper error: ${messageLabel}, Context: ${context}`,
-        template: 'error-email',
-        context: {
-          root: context,
-          time: now,
-          message: typeof message === 'object' ? message?.label : message,
-          data: Tableify(message?.data),
-          stackTrace: errorTrace
-        }
-      }).then(errorEmailResponse => {
-        this.log({label: 'Error notification email sent.', data: { mailSendResponse: errorEmailResponse }}, MineLoggerService.name);
-      }).catch(e => {
-        this.log({label: 'Failed to send error notification email', data: { mailSendResponse: e.message }}, MineLoggerService.name);
-      });
+      if (
+        !(this.messagesToIgnore.includes(messageLabel))  // the message isn't in the array of messages to skip sending emails for
+        && this.configService.get('email.system.enableErrorReportEmail')  // and error report emails are enabled
+      ) {
+        const now = new Date().toUTCString();
+        this.mailerService.sendMail({
+          to: this.configService.get('email.system.errorReportReceiver'),
+          from: this.configService.get('email.default.sender'),
+          subject: `m9sweeper error: ${messageLabel}, Context: ${context}`,
+          template: 'error-email',
+          context: {
+            root: context,
+            time: now,
+            message: typeof message === 'object' ? message?.label : message,
+            data: Tableify(message?.data),
+            stackTrace: errorTrace
+          }
+        }).then(errorEmailResponse => {
+          this.log({
+            label: 'Error notification email sent.',
+            data: { mailSendResponse: errorEmailResponse }
+          }, MineLoggerService.name);
+        }).catch(e => {
+          this.log({
+            label: 'Failed to send error notification email',
+            data: { mailSendResponse: e.message }
+          }, MineLoggerService.name);
+        });
+      }
+    } catch (e) {
+      console.log('An error occurred while attempting to log an error message. Information: ', message, trace, context, { loggingError: e });
```

Comment:
Why console.log here?

Reply:
Because this is in the logging service. If all else fails and it can't log the official way, then there'll at least be something present.

Reply:
haha oh right :D